src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,965 @@
+     * @throws DateTimeException if the value of any field is out of range,
+     *                           or if the day-of-month is invalid for the month-year
+     */
+    public static InternationalFixedDate yearDay(final int prolepticYear) {
+        return createYearDay(prolepticYear);
+    }
+
+    //-----------------------------------------------------------------------
+    /**
+     * Obtains a {@code InternationalFixedDate} from a temporal object.
+     * <p/>
+     * This obtains a date in the International fixed calendar system based on the specified temporal.
+     * A {@code TemporalAccessor} represents an arbitrary set of date and time information,
+     * which this factory converts to an instance of {@code InternationalFixedDate}.
+     * <p/>
+     * The conversion typically uses the {@link ChronoField#EPOCH_DAY EPOCH_DAY}
+     * field, which is standardized across calendar systems.
+     * <p/>
+     * This method matches the signature of the functional interface {@link TemporalQuery}
+     * allowing it to be used as a query via method reference, {@code InternationalFixedDate::from}.
+     *
+     * @param temporal the temporal object to convert, not null
+     * @return the date in the International fixed calendar system, not null
+     * @throws DateTimeException if unable to convert to a {@code InternationalFixedDate}
+     */
+    public static InternationalFixedDate from(final TemporalAccessor temporal) {
+        if (temporal instanceof InternationalFixedDate) {
+            return (InternationalFixedDate) temporal;
+        }
+
+        return InternationalFixedDate.ofEpochDay(temporal.getLong(ChronoField.EPOCH_DAY));
+    }
+
+    /**
+     * Obtains a {@code InternationalFixedDate} representing a date in the International fixed calendar
+     * system from the proleptic-year and day-of-year fields.
+     * <p/>
+     * This returns a {@code InternationalFixedDate} with the specified fields.
+     * The day must be valid for the year, otherwise an exception will be thrown.
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param dayOfYear     the International fixed day-of-year, from 1 to 366
+     * @return the date in International fixed calendar system, not null
+     * @throws DateTimeException if the value of any field is out of range, or if the day-of-year is invalid for the year
+     */
+    static InternationalFixedDate ofYearDay(final int prolepticYear, final int dayOfYear) {
+        InternationalFixedChronology.YEAR_RANGE.checkValidValue(prolepticYear, ChronoField.YEAR_OF_ERA);
+        InternationalFixedChronology.DAY_OF_YEAR_RANGE.checkValidValue(dayOfYear, ChronoField.DAY_OF_YEAR);
+
+        if (dayOfYear == InternationalFixedChronology.DAYS_IN_YEAR + 1 && !InternationalFixedChronology.INSTANCE.isLeapYear(prolepticYear)) {
+            throw new DateTimeException("Invalid Year Day: " + prolepticYear + '/' + dayOfYear);
```

Comment:
error message should be changed to reference actual problem of "not a leap year".  See existing implementations.